File: Clientes OpenInvoicePeru UBL 2.1/OpenInvoicePeru.ApiClientCSharp/RestHelper.cs
﻿using RestSharp;

namespace OpenInvoicePeru.ApiClientCSharp
{
    public static class RestHelper<TRequest, TResponse>
        where TRequest : class
        where TResponse : class, new()
    {
        public static TResponse Execute(string metodo, TRequest request)
        {
            var client = new RestClient("http://localhost/OpenInvoicePeruUBL21/api");

            var restRequest = new RestRequest(metodo, Method.POST)
            {
                RequestFormat = DataFormat.Json
            };

            restRequest.AddBody(request);

            var restResponse = client.Execute<TResponse>(restRequest);
            return restResponse.Data;
        }

    }
}

File: Clientes OpenInvoicePeru UBL 2.1/OpenInvoicePeru.ApiClientVisualBasic/RestHelper.vb
Imports RestSharp

Public Class RestHelper(Of TRequest As Class, TResponse As {Class, New})

    Public Shared Function Execute(metodo As String, request As TRequest) As TResponse
        Dim client As New RestClient("http://mirandaef.azurewebsites.net/api")

        Dim restRequest As New RestRequest(metodo, Method.POST) With {
                .RequestFormat = DataFormat.Json
                }

        restRequest.AddBody(request)

        Dim restResponse = client.Execute(Of TResponse)(restRequest)
        Return restResponse.Data
    End Function

End Class